From 546ec13c0df3ecbd60e66213f4aef49f8f39dfeb Mon Sep 17 00:00:00 2001
From: Szymon Janc <szymon.janc@tieto.com>
Date: Thu, 9 Apr 2015 14:11:29 +0200
Subject: [PATCH] profiles/network: Fix not being able to initiate connection

This was due to incorrect use of bt_uuid_to_uuid128 where source
and destination parameters were reversed.
---
 profiles/network/connection.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/profiles/network/connection.c b/profiles/network/connection.c
index 2b07771..77e0a34 100644
--- a/profiles/network/connection.c
+++ b/profiles/network/connection.c
@@ -293,7 +293,7 @@ static DBusMessage *local_connect(DBusConnection *conn,
 
 	id = get_pan_srv_id(svc);
 	bt_uuid16_create(&uuid16, id);
-	bt_uuid_to_uuid128(&uuid128, &uuid16);
+	bt_uuid_to_uuid128(&uuid16, &uuid128);
 
 	if (bt_uuid_to_string(&uuid128, uuid_str, MAX_LEN_UUID_STR) < 0)
 		return btd_error_invalid_args(msg);
@@ -447,7 +447,7 @@ static gboolean network_property_get_uuid(const GDBusPropertyTable *property,
 		return FALSE;
 
 	bt_uuid16_create(&uuid16, nc->id);
-	bt_uuid_to_uuid128(&uuid128, &uuid16);
+	bt_uuid_to_uuid128(&uuid16, &uuid128);
 	bt_uuid_to_string(&uuid128, uuid_str, MAX_LEN_UUID_STR);
 
 	dbus_message_iter_append_basic(iter, DBUS_TYPE_STRING, &uuid_str);
-- 
2.4.2

